web_src/js/utils/dom.ts

```diff
@@ -301,10 +301,17 @@ export function replaceTextareaSelection(textarea: HTMLTextAreaElement, text: st
 }
 
 // Warning: Do not enter any unsanitized variables here
-export function createElementFromHTML(htmlString: string): HTMLElement {
+export function createElementFromHTML<T extends HTMLElement>(htmlString: string): T {
```

Comment:
With `extends`, the type parameter becomes required, right? I would prefer it to be optional.

Does `T extends HTMLElement = HTMLElement` work?

Reply:
`T extends HTMLElement` implies `HTMLElement` by default, if I understand correctly.

![image](https://github.com/user-attachments/assets/b76d4bba-2ca2-4889-a02a-224f6b25d21c)

Reply:
If there is no `tsc` error at the call site, it should be fine, please check.

Reply:
Ok then. I wasn't aware the `extends` also implies a default, but as per your screenshot, it seems to work.